Caltech-IPAC/firefly
# FIREFLY-1376: Convert textarea to JoyUI

Fixes [FIREFLY-1376](https://jira.ipac.caltech.edu/browse/FIREFLY-1376)

- Converted InputAreaFieldView to use joyui Textarea
  - exposed common code in InputFieldView to use here
  - cleaned up lot of styling props and css file no longer needed
- Updated usages of InputAreaFieldView (see testing), cleaning up old props

Note: joyui Textarea uses minRows and maxRows to control number of lines shown in textarea initially and to control maximum lines it can expand to. This means setting minRows=maxRows will create a fixed height textarea which is what old`rows` prop was controlling.

### Extras:
- [x] improve spacing and layout of CatalogSearch panel
- [x] add spacing guidelines to documentation

## Testing
https://fireflydev.ipac.caltech.edu/firefly-1376-textarea/firefly/

Check following:
- Catalog search panel 
   - change search method to polygon -> coordinates **textarea**
   - additional SQL constraints **textarea**
- TriViewPanel
   - table options -> advanced filters tab -> constraints **textarea**
   - chart filters -> filters **textarea**
   - select an area in coverage -> binoculors btn -> search refinement option -> polygon radio btn -> coordinates **textarea** (yellow colored because it's connected)
- TAP search -> Edit ADQL -> ADQL **textarea** (with prismjs coloring - background as well as text)

Extras:
Visually inspect the catalog selection panel (or by web inspector), there should be no unnecessary paddings. And layout shouldn't jump vertically when changing catalog or search method.
